src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandleExtensions.cs

```diff
@@ -0,0 +1,30 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Provides extension methods to operate with GC handles.
+    /// </summary>
+    public static class GCHandleExtensions
+    {
+        // The following methods are strongly typed generic specifications only on
+        // PinnedGCHandles with correct type.
+
+        /// <summary>
+        /// Retrieves the address of string data in a <see cref="PinnedGCHandle{T}"/> of array.
+        /// </summary>
+        /// <returns>The address of the pinned array.</returns>
+        [CLSCompliant(false)]
+        public static unsafe T* GetAddressOfArrayData<T>(this PinnedGCHandle<T[]?> handle)
+            => (T*)handle.GetAddressOfObjectData();
+
+        /// <summary>
+        /// Retrieves the address of string data in a <see cref="PinnedGCHandle{T}"/> of <see cref="string"/>.
+        /// </summary>
+        /// <returns>The address of the pinned <see cref="string"/>.</returns>
+        [CLSCompliant(false)]
+        public static unsafe char* GetAddressOfStringData(this PinnedGCHandle<string?> handle)
+            => (char*)handle.GetAddressOfObjectData();
```

Comment:
Should these be nullable oblivious? Given there's no covariance, annotating either with nullable or non-nullable will give warning for the handles annotated with the opposite.

Reply:
We suggested making these nullable during API review under the assumption that that would allow both without warnings. If that's not actually the case (as in, calling this with a non nullable `T` still produces a warning), making these nullable oblivious makes sense to me 🤔

Reply:
What about only making the parameter nullable oblivious? Would that be doable?
As in:
```csharp
public static unsafe T* GetAddressOfArrayData<T>(
#nullable disable
    this PinnedGCHandle<T[]?> handle)
#nullable restore
    => (T*)handle.GetAddressOfObjectData();
```
So we can preserve the nullability annotation on the returned pointer?

Reply:
While I'm not seeing any behavioral difference for type inference and nullability warning, the compiled metadata are different. Agreed that nullable oblivious range should be as short as possible.